components/Header/Header.jsx

```diff
@@ -17,8 +17,9 @@ const Header = ({
   headerTextColor,
 }) => {
   const [isModalVisible, setModalVisible] = useState(false);
-  const handleLogout = () => setModalVisible(true);
-
+  const handleLogout = useCallback(() => {
+    setModalVisible(true);
+  }, []);
   return (
     <>
```

Comment:
create a function for these styles 

const selectAppropriateStyle = (className, color) => (
[
                styles[className],
                {color: headerTextColor || theme.colors[color]},
])

then   
<Text style={selectAppropriateStyle("headerButtonText", "primary_green")}>